src/connection.ts

```diff
@@ -95,14 +107,16 @@ export class StableWSConnection<
 
   /**
    * connect - Connect to the WS URL
-   *
+   * the default 15s timeout allows between 2~3 tries
    * @return {ConnectAPIResponse<ChannelType, CommandType, UserType>} Promise that completes once the first health check message is received
    */
-  async connect() {
+  async connect(timeout = 15000) {
```

Comment:
```suggestion
  async connect(options = {timeout: 15000}) {
```
what do you think of an object as argument to give a little bit more clarity to what it does?

Reply:
yea i prefer it this way since it's clear to understand the param on the spot, there are more than one place that can benefit from this refactor so let's keep it for another PR